src/gui/editorwidgets/core/qgseditorwidgetwrapper.h

```diff
@@ -277,6 +277,18 @@ class GUI_EXPORT QgsEditorWidgetWrapper : public QgsWidgetWrapper
      */
     virtual void updateConstraintWidgetStatus();
 
+  protected:
+
+    /**
+     * mFieldIdx the widget feature field id
+     */
+    int mFieldIdx;
+
+    /**
+     * mFeature the current feature
+     */
+    QgsFeature mFeature;
```

Comment:
That means the attributes of the feature can be changed without appropriate notifications being sent? I'd prefer to avoid having this directly writable to subclasses and make (protected?) writers instead.
The adding docs is much appreciated, but should include information about which state this feature has (i.e. the feature which was set originally or the current state of the feature on the form.